clang-tidy-cache

```diff
@@ -922,12 +923,19 @@ class ClangTidyCache(object):
             self._caches.append(ClangTidyGcsCache(log, opts))
 
         if not self._caches:
-            self._caches.append(ClangTidyLocalCache(log, opts))
+            self._add_local_cache(log, opts)
+
+    # --------------------------------------------------------------------------
+    def _add_local_cache(self, log, opts):
+        self._local_cache = ClangTidyLocalCache(log, opts)
+        self._caches.append(self._local_cache)
 
     # --------------------------------------------------------------------------
     def is_cached(self, digest):
         for cache in self._caches:
             if cache.is_cached(digest):
+                if self._local_cache is not None and cache != self._local_cache:
+                    self._local_cache.store_in_cache(digest)
```

Comment:
This may need to be improved, i.e. by trying to do a `get_cache_data` followed by a `store_in_cache_with_data` such that any data persisted is also pulled down?